projects/addon-doc/components/api/api-item.component.ts

```diff
@@ -56,4 +69,59 @@ export class TuiDocAPIItem<T> {
 
     @Output()
     public readonly valueChange = new EventEmitter<T>();
+
+    public ngOnInit(): void {
+        this.parseParams(this.activatedRoute.snapshot.queryParams);
+    }
+
+    public onValueChange(value: T): void {
+        this.value = value;
+        this.valueChange.emit(value);
+        this.setQueryParam(value);
+    }
+
+    @tuiPure
```

Comment:
No need for memoization, it is not used in a getter

Reply:
removed